static/modules/gamemanager.js

```diff
@@ -62,23 +64,38 @@ class GameManager {
     displayWidth = 960,
     displayHeight = 540
   ) {
+    // the canvas for drawing
     this.canvas = document.createElement("canvas");
     this.context = this.canvas.getContext("2d");
-    this.context.imageSmoothingEnabled = false;
+    //this.context.imageSmoothingEnabled = false;
     this.canvas.width = width;
     this.canvas.height = height;
 
+    // the canvas for displaying
     this.displayCanvas = document.createElement("canvas");
     this.displayContext = this.displayCanvas.getContext("2d");
-    this.displayContext.imageSmoothingEnabled = false;
+    //this.displayContext.imageSmoothingEnabled = false;
     this.displayWidth = displayWidth;
     this.displayHeight = displayHeight;
     this.displayCanvas.width = displayWidth;
     this.displayCanvas.height = displayHeight;
 
+    // the untouched canvas for blurring before copying
+    this.blurCanvas = document.createElement("canvas");
```

Comment:
Can we draw UI elements only on the non-blurred canvas somehow?

Reply:
I could change the blend mode before the UI is drawn or something. I'll try to pick more contrasting colors or rework it somehow, and if that still looks bad, that's what I'll do.